index.js

```diff
@@ -0,0 +1,29 @@
+// index.js
+protobuf = require("protobufjs");
+
+// Object.freeze(Object.prototype);
+
+let person1 = {firstName:"John", lastName:"Doe", age:50, eyeColor:"blue"};
+let person2 = {firstCoolName:"John", lastName:"Doe", age:50, eyeColor:"blue"};
+
+// Vector 1
+let evilkey = "__proto__.firstName"
+let evilval = "evilvalue"
+protobuf.util.setProperty(person1, evilkey, evilval); 
+
+// Vector 2
+let obj = new protobuf.ReflectionObject("Test")
+obj.setParsedOption({}, evilval, evilkey);
+
+// Vector 3
+let p = `option (foo).__proto__.someprop= "somevalue";` 
+protobuf.parse(p)
```

Comment:
<div align="center"> 

### 📦🔍 Contextual Analysis CVE Vulnerability

| Severity | Finding |
| :------: | :---: |
| ![](https://raw.githubusercontent.com/jfrog/frogbot/master/resources/v2/applicableHighSeverity.png)<br> | Prototype pollution `Object.freeze` remediation was not detected, The vulnerable function protobufjs.parse is called with external input, The vulnerable function protobufjs.load(Sync) is called |

</div>

<details>
<summary> <b>Description</b> </summary>
<br>

The scanner checks whether any of the following vulnerable functions are called:

* `util.setProperty` with external input to its 2nd (`path`) or 3rd (`value`) arguments.
* `ReflectionObject.setParsedOption` with external input to its 2nd (`name`) or 3rd (`value`) arguments.
* `parse` with external input to its 1st (`source`) argument.
* `load`
* `loadSync`

The scanner also checks whether the `Object.freeze()` remediation is not present.

</details>

<details>
<summary> <b>CVE details</b> </summary>
<br>

The package protobufjs before 6.11.3 are vulnerable to Prototype Pollution which can allow an attacker to add/modify properties of the Object.prototype. This vulnerability can occur in multiple ways: 1. by providing untrusted user input to util.setProperty or to ReflectionObject.setParsedOption functions 2. by parsing/loading .proto files

</details>


<details>
<summary> <b>Remediation</b> </summary>
<br>

##### Development mitigations

Add the `Object.freeze(Object.prototype);` directive once at the beginning of your main JS source code file (ex. `index.js`), preferably after all your `require` directives. This will prevent any changes to the prototype object, thus completely negating prototype pollution attacks.

</details>


---
<div align="center">

[🐸 JFrog Frogbot](https://github.com/jfrog/frogbot#readme)

</div>